readthedocs/proxito/views/serve.py

```diff
@@ -95,6 +111,11 @@ def get(self,
         # Handle requests that need canonicalizing (eg. HTTP -> HTTPS, redirect to canonical domain)
         if hasattr(request, 'canonicalize'):
             try:
+                # A canonical redirect can be cached, if we don't have information
+                # about the version, since the final URL will check for authz.
+                if not version and self._is_cache_enabled(final_project):
```

Comment:
I think we can always cache on these redirects, even if we have a version and that one is private. Since we don't check for authz after all these (basically the requests are the same for all users).